Mathlib/Data/Polynomial/Module.lean

```diff
@@ -196,7 +197,11 @@ theorem smul_apply (f : R[X]) (g : PolynomialModule R M) (n : ℕ) :
     exacts [rfl, (zero_smul R _).symm]
 #align polynomial_module.smul_apply PolynomialModule.smul_apply
 
+-- Porting note: typeclass inference goes haywire in the next declaration.
+-- One way to prune the bad branches is:
+attribute [-instance] IsDomain.toCancelCommMonoidWithZero
```

Comment:
This looks like more of the thing we're trying to remove in this PR; is it worth mentioning `etaExperiment` or `2074` in the comment for later discoverability?

Also, should this be

```suggestion
-- Porting note: typeclass inference goes haywire in the next declaration.
-- One way to prune the bad branches is:
attribute [-instance] IsDomain.toCancelCommMonoidWithZero in
```
to limit it to the one declaration?